Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -1098,6 +1098,80 @@ extension EventLoopFuture {
         }
         return CallbackList()
     }
+
+    #if swift(>=5.7)
+    /// Adds an observer callback to this `EventLoopFuture` that is called when the
+    /// `EventLoopFuture` has a success result, asserting a failure if it encounters an error.
+    ///
+    /// - parameters:
+    ///     - callback: The callback that is called with the successful result of the `EventLoopFuture`.
+    @inlinable
+    @preconcurrency
```

Comment:
As these are new methods they don't need `@preconcurrency` and also no special treatment in Swift 5.6. We can therefore just have a single declaration and implementation. 

I'm with @Lukasa though and we should make these methods return an `EventLoopFuture` instead. Therefore we will drop the `@Sendable` function argument all together and AFAICT don't have to deal with `Sendable` at all.

Reply:
Okay i see, that is more for the older methods that were implemented before some language features were introduced,  I will work on addressing the needed changes. Thank you all.